### PR TITLE
Spectra container class

### LINKED ISSUE #439: Unifying RSS and Cube classes
In principle, many (if not most/all) corrections could be applied equally well to `RSS` and `Cube` objects. Usually, their `intensity` and `variance` attributes are simply treated as a collection of spectra. I will not insist on `specutils.Spectrum1D` (oops, I just did ;^D), but I think both objects have many attributes and methods in common, and that should be properly capture in the architecture. What about the following options?
**Option 1**
```python
class SpectraContainer(DataContainer):
    wavelength
    intensity
    variance
    get_rss_intensity()
    get_rss_variance()

class RSS(SpectraContainer):
    def get_rss_intensity(self):
        return self.intensity
    def get_rss_variance(self):
        return self.variance

class Cube(SpectraContainer):
    def get_rss_intensity(self):
        return np.reshape(self.intensity, (self.intensity.shape[0], self.intensity.shape[1]*self.intensity.shape[2]).T
    def get_rss_variance(self):
        return np.reshape(self.variance, (self.variance.shape[0], self.variance.shape[1]*self.variance.shape[2]).T
```
**Option 2**
Instead of a common parent (`SpectraContainer`), `Cube` could be a child from `RSS` and override `get_rss_*`

To be honest, I am more inclined towards Option 1, preparing for the time where other data types, such as `astropy.CCDData` (oops again ;^D) hang from `DataContainer`.

### PR DESCRIPTION
**Pull request by paranoya**
fix #439

### CODE REVIEW

**Comment by paranoya** [comment]
Please have a look to the implementation. In particular, I am not sure about the right way to handle the constructor parameters. Now I'm just using `**kwargs`, but I wonder whether I should treat compulsory arguments (`wavelength` and `intensity`) differently from optional ones (e.g. `variance`).

A related thing would be validation (e.g. units). It should only take place at the relevant class (e.g. `DataContainer` for `intensity`, `SpectraContainer` for `wavelength`, `Cube` and `RSS` when testing the consistency of array shapes, etc.). It can be done with `assert` if we use `**kwargs`, but I wonder whether it would be more elegant if we used type hints.

Finally, I am now using [Black](https://github.com/psf/black) to fix code format, and I am considering moving towards [Ruff](https://docs.astral.sh/ruff/) for consistency with `astropy`:
<https://docs.astropy.org/en/latest/development/codeguide.html#coding-style-conventions>
The point is that Black has introduced a few minor modifications (double quotes, line breaks) that you may or may not like. I am fully convinced about converging towards `astropy`, but I'd be happy to hear your thoughts.

**Comment by paranoya** [comment]
Oops, I've also noticed that I just used my `sky-based_calibration.ipynb`  tutorial for testing, but my modifications break everything else. I'll change the PR to "Draft" state and fix it. It shouldn't take long (I hope), but please don't wait for me to send comments.

**Comment by paranoya** [comment]
I quick-fixed it by allowing the creation of empty `Cube`s (and hence `SpectraContainer`) when no `wavelength` is passed to the constructor. Honestly, I find it a somewhat dirty hack...

File: src/pykoala/rss.py [hunk 1]
```diff
@@ -50,18 +50,18 @@ class RSS(DataContainer):
...
-        # Intialise base class
-        super().__init__(intensity=intensity, variance=variance, info=info, log=log)
-        # Specific RSS attributes
-        self.wavelength = wavelength
+    @property
+    def rss_variance(self):
+        return self._variance
+
+    def __init__(self, **kwargs):
+        assert ('wavelength' in kwargs)
```
Comment (PabloCorcho) [inline src/pykoala/rss.py:62]: Shouldn't this check be part of the SpectraContainer constructor?

File: src/pykoala/data_container.py [hunk 2]
```diff
@@ -361,58 +389,157 @@ class DataContainer(object):
...
+        """Array with the counts/surface brightness/... at each pixel."""
+        return self._intensity
+
+    @intensity.setter
+    def intensity(self, value):
+        self._intensity = value
+
+    @intensity.deleter
+    def intensity(self):
+        del self._intensity
```
Comment (PabloCorcho) [inline src/pykoala/data_container.py:419]: I guess it's fine, but do we need to include deleters? In which situation are they going to be used?

**Review by PabloCorcho — APPROVED** [review]
This can be safely merged

**Pull request merged**